content/sensu-go/5.17/reference/secrets.md

```diff
@@ -0,0 +1,213 @@
+Sensu backends cache fetched secrets in memory, with no persistence to a Sensu datastore or file on disk.
+Secrets provided via a "lease" with a "lease duration" are deleted from Sensu's in-memory cache after the configured number of seconds, prompting the Sensu backend to request the secret again.
+
+Secrets are only transmitted over a TLS websocket connection.
+Unencrypted connections must not transmit privileged information.
+For agent-side resources, enable TLS/mTLS.
+
+Secrets are only exposed to Sensu services like environment variables and are automatically redacted from all logs, the API, and the Sensu dashboard.
+ 
+## Secret specification
+
+### Top-level attributes
+
+type         | 
+-------------|------
+description  | Top-level attribute that specifies the resource type. For secrets configuration, the type should always be `Secret`.
+required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
+type         | String
+example      | {{< highlight shell >}}"type": "Secret"{{< /highlight >}}
+
+api_version  | 
+-------------|------
+description  | Top-level attribute that specifies the Sensu API group and version. For secrets configuration in this version of Sensu, the `api_version` should always be `secrets/v1`.
+required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
+type         | String
+example      | {{< highlight shell >}}"api_version": "secrets/v1"{{< /highlight >}}
+
+metadata     |      |
+-------------|------
+description  | Top-level scope that contains the secret's `name` and `namespace`.
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"metadata": {
+  "name": "sensu-ansible-token",
+  "namespace": "default"
+}
+{{< /highlight >}}
+
+spec         | 
+-------------|------
+description  | Top-level map that includes secrets configuration [spec attributes][12].
+required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"spec": {
+  "id": "ANSIBLE_TOKEN",
+  "provider": "ansible_vault"
+}
+{{< /highlight >}}
+
+### Metadata attributes
+
+name         |      |
+-------------|------
+description  | The name for the secret that is used internally by Sensu.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"name": "sensu-ansible-token"{{< /highlight >}}
+
+namespace    |      |
+-------------|------
+description  | Provider name used internally by Sensu.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"namespace": "default"{{< /highlight >}}
+
+### Spec attributes
+
+id           | 
+-------------|------ 
+description  | The identifying key for the provider to retrieve the secret.
+required     | true
+type         | String
+default      | `false`
+example      | {{< highlight shell >}}"id": "ANSIBLE_TOKEN"{{< /highlight >}}
```

Comment:
Since the secret example being used is referencing a vault provider, we should use the syntax one would use to retrieve a vault secret.
```suggestion
example      | {{< highlight shell >}}"id": "secret/ansible#token"{{< /highlight >}}
```